source/fe/mapping_q.cc

```diff
@@ -290,12 +290,23 @@ MappingQ<dim, spacedim>::transform_unit_to_real_cell(
   const typename Triangulation<dim, spacedim>::cell_iterator &cell,
   const Point<dim>                                           &p) const
 {
-  return Point<spacedim>(internal::evaluate_tensor_product_value(
-    polynomials_1d,
-    make_const_array_view(this->compute_mapping_support_points(cell)),
-    p,
-    polynomials_1d.size() == 2,
-    renumber_lexicographic_to_hierarchic));
+  if (polynomial_degree > 1)
```

Comment:
In my opinion `if (polynomial_degree == 1)` logic is easier to read, like we do in `tensor_product_point_kernels.h` but that's a matter of taste.